commit b75a736f3dce31f76bb9b5a8d6cccfe8c5a491ac
Author: Richard Gill <richard@rgill.co.uk>
Date:   Mon Dec 8 09:16:52 2025 +0000

    Remove Joist configuration and pipeline files

diff --git b/.joist/config.pkl a/.joist/config.pkl
new file mode 100644
index 0000000..f023a7f
--- /dev/null
+++ a/.joist/config.pkl
@@ -0,0 +1,25 @@
+// See the comment on `.joist/pipelines/check.pkl` for information about `amends`
+amends "https://joist.fsn1.your-objectstorage.com/pkl/Config.pkl"
+
+// RunFrom determines whether this pipeline should run from local or from remote.
+// When set to local, it allows `local` users to complete it's pipeline. If `remote`,
+// the pipeline needs to be executed from a `remote` server [not yet available].
+// This is an example of using a conditional in Pkl.
+// `env.BranchName` is set by `Environment.pkl` (check pkl/Environment.pkl above), which is imported by `Config.pkl`.
+RunFrom = if (env.BranchName == "main") "remote" else "local"
+
+// Version determines which Joist version this file and the pipelines are written for.
+Version = "0.0.1"
+
+// API Key is your API key; here's how it can be used to read from your shell's env. You're also welcome
+// to hard code this value (`ApiKey = "foo"`). As long as `ApiKey` exists, Joist will use it.
+ApiKey = read("env:JOIST_API_KEY")
+
+// Repo is used to determine which repository you're actively working on. This needs to be filled in order
+// to match with the webhooks Github sends to Joist. Additionally, you need to ensure that the repository
+// is added to your user on https://joist.dev/
+Repo {
+    Owner = "richardgill"
+    Name = "patchy"
+    Provider = "github"
+}
diff --git b/.joist/pipelines/local.pkl a/.joist/pipelines/local.pkl
new file mode 100644
index 0000000..1f63a45
--- /dev/null
+++ a/.joist/pipelines/local.pkl
@@ -0,0 +1,27 @@
+amends "https://joist.fsn1.your-objectstorage.com/pkl/Pipeline.pkl"
+
+import "../shared.pkl"
+
+stages = new Listing<Stage> {
+  new {
+    name = "Lint"
+    system = shared.main
+    actions = new Listing {
+      "pnpm run check"
+    }
+  }
+  new {
+    name = "Typecheck"
+    system = shared.main
+    actions = new Listing {
+      "pnpm run typecheck"
+    }
+  }
+  new {
+    name = "Test"
+    system = shared.main
+    actions = new Listing {
+      "pnpm run test"
+    }
+  }
+}
diff --git b/.joist/shared.pkl a/.joist/shared.pkl
new file mode 100644
index 0000000..0347fc0
--- /dev/null
+++ a/.joist/shared.pkl
@@ -0,0 +1,9 @@
+import "https://joist.fsn1.your-objectstorage.com/pkl/Pipeline.pkl" as Pipeline
+
+main = new Pipeline.LocalSystem {
+  allowed_bins = new Listing { "pnpm"; "npm"; "node"; "mise" }
+  use_default_bins = true
+  setup = new Listing {
+    "pnpm install"
+  }
+}
\ No newline at end of file
diff --git b/lefthook.yml a/lefthook.yml
index d5df01c..add8397 100644
--- b/lefthook.yml
+++ a/lefthook.yml
@@ -15,3 +15,9 @@ pre-push:
     biome:
       glob: "*.{js,ts,cjs,mjs,d.cts,d.mts,jsx,tsx,json,jsonc}"
       run: pnpm exec biome check --error-on-warnings --diagnostic-level=warn --no-errors-on-unmatched --files-ignore-unknown=true {push_files}
+
+post-commit:
+  commands:
+    joist-run:
+      run: |
+        tmux new-session -d -s jcli 'jcli run . > jcli.log 2>&1 && tmux kill-session -t jcli'
